### ISSUE TITLE
Feature request: Reassign hotkey to the parent group (-No Group-)

### ISSUE THREAD

**Issue by pyxelr** [sabrogden/Ditto#163]
Currently, we can "_Go to the Parent Group_" using the `Backspace` hotkey (as mentioned [in the Wiki](https://github.com/sabrogden/Ditto/wiki#key-bindings)). Is it possible to reassign this key?

I am using `Ctrl + 1`  to open a specific subgroup, and then my hand has to reach quite far to move back to the parent group.
Unfortunately, right-clicking on `-No Group-` and selecting `Properties` does not open any window to assign a key there.

**Comment by Chealer** [comment]
Are you talking about a shortcut for the parent group, or for the top/root/null group?

**Comment by pyxelr** [comment]
I am talking about the default root group marked with (1) on this screenshot:

<img width="488" height="842" alt="Image" src="https://github.com/user-attachments/assets/aa435d28-a2a6-465f-9081-c74c8ec66015" />

- (1) can be opened using `Backspace` and this hotkey cannot be changed
- (2) can be opened using any hotkey we specify

---

Anyway, I have switched from using groups in Ditto to espanso, so we can close this thread if no one else is interested.

**Comment by Chealer** [comment]
Right. Sorry, but what I meant is that any group can have a subgroup. If you are in subsubgroup 1, are you talking about a shortcut which goes to the root, or only to the parent (subgroup 1)?

> Anyway, I have switched from using groups in Ditto to espanso, so we can close this thread if no one else is interested.

I am not sure what you mean; this is a ticket in Ditto's ITS.
Just to save some time for other curious readers: While espanso may be interesting, it is not a clipboard manager and is far from being an equivalent to Ditto, although both tools can address a minority of Ditto's use cases.

<details>
  <summary>License</summary>
This comment and any other from myself on this ticket is offered under the terms of CC0 1.0.
</details>

**Comment by pyxelr** [comment]
I mean the keyboard shortcut that moves from a subgroup back to its parent, which in this case is the root group named `-No Group-`, as shown in the screenshot in my previous comment.

**Comment by Chealer** [comment]
Thanks

In that case, please:
1. amend/clarify the ticket's body
2. retitle. If you are asking to make the shortcut customizable, I suggest something like:
   > Keyboard shortcut to navigate to parent group cannot be customized

**Comment by pyxelr** [comment]
1. Ticket body is clear to me. I am only talking about the core root group.
2. I think you are getting me wrong. Parent group of subsubgroup1 would be subgroup1, but Ditto makes it possible to reassign a hotkey for subgroup1. I am talking about the core root level group of Ditto called `-No Group-`. 

Sorry, but if something is not clear in this issue, I suggest an [actual contributor](https://github.com/sabrogden/Ditto/graphs/contributors) to ask me for a clarification.

**Comment by Chealer** [comment]
> 1. Ticket body is clear to me. I am only talking about the core root group.

No, it talks about both the root group and the parent group. Ditto itself is confusing enough by calling the root/top group "No Group"; we won't understand your request if you confuse a group's parent with the root.

> Sorry, but if something is not clear in this issue, I suggest an [actual contributor](https://github.com/sabrogden/Ditto/graphs/contributors) to ask me for a clarification.

Sorry, but this "project" has badly limited means, and your ticket does not comply. If you don't mind, please discard it, and let actual Ditto users and contributors report your issues.

**Comment by vertigo220** [comment]
I would say it should allow both changing the key for moving up to the current group's parent (e.g. instead of backspace, allow the user to configure a different key) _as well as_ allowing a hotkey to be assigned to the top/root (-No Group-) "group" as currently you can assign hotkeys to other groups (subgroups) but not to the main one.

It should also be noted that I didn't even realize until I saw it mentioned in an issue that hotkeys could be assigned to groups. I actually came here to request this, since there's no settings for group hotkeys in settings and I didn't realize you had to go to their individual properties and set them there. It makes sense to do it that way, but there should be a note regarding this in the hotkeys section of the main settings, to guide users.

Finally, and perhaps this should be a separate issue, I'd really like the ability to set a hotkey to toggle groups as well as one to open/close the groups selection. This way, each group wouldn't require its own hotkey to be remembered, and it would be easy to use just the keyboard to launch Ditto, switch groups, and search.

**Comment by Chealer** [comment]
@vertigo220 wrote:
> Finally, and perhaps this should be a separate issue, I'd really like the ability to set a hotkey to toggle groups as well as one to open/close the groups selection.

What do you mean by "opening the groups selection", if not what <kbd>Return</kbd> does?
I would agree with making <kbd>Ctrl</kbd>+<kbd>G</kbd> a toggle rather than only _opening_ the panel, but that should be tracked in another ticket.

**Comment by vertigo220** [comment]
> What do you mean by "opening the groups selection"

I mean what Ctrl+G does, which I didn't even know of that hotkey since it's not in the settings anywhere, at least not that I've seen. It should be a customizable hotkey, both to make it more apparent and to allow it to be set to whatever the user prefers.

**Comment by Chealer** [comment]
@vertigo220: If that's what you call "opening the groups selection", then:
1. I am not sure what you mean by "toggle groups".
2. "The groups selection" can be opened with <kbd>Ctrl</kbd>+<kbd>G</kbd>.
3. "The groups selection" can be closed with <kbd>Escape</kbd>.

**Comment by vertigo220** [comment]
Toggle meaning switch to the next, the next, and so on. Cycle is a better word. Similarly to how there are different hotkeys to paste any of the last 10 items, but there's one hotkey to open Ditto and then to cycle through the entries, you can assign each group its own hotkey, but there should be one that can be hit repeatedly to move through them in similar fashion. 

And instead of having to use two separate keys to open/close grouos, neither of which are configurable or, by all appearances, even made known in the program, allowing the user to choose a hotkey to open AND close.

**Comment by Chealer** [comment]
@vertigo220 wrote:
> Toggle meaning switch to the next, the next, and so on.

I do not speak English natively, but "toggle" (almost?) always means alternating between 2 states; most people would not understand what you meant by ”toggle groups”.

> Cycle is a better word.

Indeed

> Similarly to how there are different hotkeys to paste any of the last 10 items, but there's one hotkey to open Ditto and then to cycle through the entries, you can assign each group its own hotkey, but there should be one that can be hit repeatedly to move through them in similar fashion.

That would require establishing an order between all groups. But I am not convinced such a shortcut is a good idea.

> And instead of having to use two separate keys to open/close grouos, neither of which are configurable or, by all appearances, even made known in the program, allowing the user to choose a hotkey to open AND close.

The shortcut is indicated is the _Groups_ submenu of an item’s contextual menu. The problem is that placing _View Groups_ in a clip’s contextual menu is counterintuitive, as _View Groups_ does not affect any specific clip.

As I wrote yesterday, I would agree with making _View Groups_ a toggle, but this should be tracked in a different ticket. I think it would be acceptable to create a wider ticket tracking the efficiency of groups navigation in general (moving up, laterally or down). Solving specific related issues in isolation, like creating a shortcut to display groups and then another one to hide them, may not create an optimal UX.

**Comment by vertigo220** [comment]
> That would require establishing an order between all groups. But I am not convinced such a shortcut is a good idea.

No, it wouldn't. It would just need to go to the next group every time it's pressed. The only order that would need to e defined is the order of the groups, which already exists.

> The shortcut is indicated is the _Groups_ submenu of an item’s contextual menu. The problem is that placing _View Groups_ in a clip’s contextual menu is counterintuitive, as _View Groups_ does not affect any specific clip.

I agree that is not where it should be. But neither is the current location. It makes no sense to look for it there, for the reason you stated. It would make far more sense to be in the keyboard shortcuts page in settings, and possibly in a context menu when right clicking on the folder/group icon as well.

**Comment by Chealer** [comment]
@vertigo220 wrote:
> > That would require establishing an order between all groups. But I am not convinced such a shortcut is a good idea.
> 
> No, it wouldn't. It would just need to go to the next group every time it's pressed.

I did not mean each user would need to order groups. What I meant is that the Ditto software needs to establish which of the groups is first, which is second, third, etc. There are many possibilities (alphabetical, chronological, etc), but Ditto would need to know which group is next, and whatever algorithm it would choose would likely not match each user’s wishes.
 
> > The shortcut is indicated is the _Groups_ submenu of an item’s contextual menu. The problem is that placing _View Groups_ in a clip’s contextual menu is counterintuitive, as _View Groups_ does not affect any specific clip.
> 
> I agree that is not where it should be. But neither is the current location. It makes no sense to look for it there, for the reason you stated.

I'm afraid there is confusion here. What I described *is* the current location.

> It would make far more sense to be in the keyboard shortcuts page in settings, and possibly in a context menu when right clicking on the folder/group icon as well.

I am skeptical that this warrant a customizable keyboard shortcut, but again, let's not make this ticket more confusing than it already is; unless its creator agrees to widen the scope, please discuss that in a dedicated ticket.
